pkg/config/model/viper.go

```diff
@@ -526,13 +526,6 @@ func (c *safeConfig) MergeConfig(in io.Reader) error {
 	return c.Viper.MergeConfig(in)
 }
 
-// MergeConfigOverride wraps Viper for concurrent access
-func (c *safeConfig) MergeConfigOverride(in io.Reader) error {
-	c.Lock()
-	defer c.Unlock()
-	return c.Viper.MergeConfigOverride(in)
-}
-
```

Comment:
Any particular reason to remove this function ?

Reply:
The only caller to this method was removed by https://github.com/DataDog/datadog-agent/pull/21223, and this PR is a follow-up to that, building on changes it made. There's undesirable behavior in `MergeConfigOverride`, namely that it removes source information from the config, that this feature work is improving by making config modifications more fine-grained.

Reply:
If it's on purpose then perfect for me :)